x-pack/legacy/plugins/lens/public/indexpattern_plugin/change_indexpattern.tsx

```diff
@@ -0,0 +1,126 @@
+  label: string;
+}
+
+export function ChangeIndexPattern({
+  indexPatterns,
+  currentIndexPatternId,
+  onChangeIndexPattern,
+  trigger,
+  layer,
+  selectableProps,
+}: {
+  trigger: ChangeIndexPatternTriggerProps;
+  indexPatterns: IndexPatternPrivateState['indexPatterns'];
+  onChangeIndexPattern: (newId: string) => void;
+  currentIndexPatternId?: string;
+  layer?: IndexPatternLayer;
+  selectableProps?: EuiSelectableProps;
+}) {
+  const [isPopoverOpen, setPopoverIsOpen] = useState(false);
+  const [selectedID, setSelectedID] = useState(
+    layer ? layer.indexPatternId : currentIndexPatternId
+  );
+
+  const indexPatternList = Object.values(indexPatterns).map(indexPattern => ({
+    ...indexPattern,
+    isTransferable: layer ? isLayerTransferable(layer, indexPattern) : undefined,
+  }));
+
+  const createTrigger = function() {
+    const { label, ...rest } = trigger;
+    return (
+      <EuiButtonEmpty
+        flush="left"
+        className="eui-textTruncate"
+        size="xs"
+        onClick={() => setPopoverIsOpen(!isPopoverOpen)}
+        {...rest}
+      >
+        {label}
+      </EuiButtonEmpty>
+    );
+  };
+
+  return (
+    <>
+      <EuiPopover
+        button={createTrigger()}
+        isOpen={isPopoverOpen}
+        closePopover={() => setPopoverIsOpen(false)}
+        className="eui-textTruncate"
+        anchorClassName="eui-textTruncate"
+        display="block"
+        panelPaddingSize="s"
+        ownFocus
+      >
+        <div style={{ minWidth: 300 }}>
+          <EuiSelectable
+            {...selectableProps}
+            searchable
+            singleSelection="always"
+            options={indexPatternList.map(indexPattern => ({
+              id: indexPattern.id,
+              label: indexPattern.title,
+              checked: indexPattern.id === selectedID ? 'on' : undefined,
+              append:
+                indexPattern && indexPattern.isTransferable !== false ? (
+                  undefined
+                ) : (
+                  <EuiIconTip
+                    color="warning"
+                    type="minusInCircle"
+                    content={i18n.translate(
+                      'xpack.lens.indexPattern.lossyIndexPatternSwitchDescription',
+                      {
+                        defaultMessage:
+                          'Not all operations are compatible with this index pattern and will be removed on switching.',
+                      }
+                    )}
+                  />
+                ),
+            }))}
+            onChange={choices => {
+              // @ts-ignore
+              const newSelectedID = choices.filter(option => option.checked)[0].id;
+              onChangeIndexPattern(newSelectedID);
+              setSelectedID(newSelectedID);
+              setPopoverIsOpen(false);
+            }}
+            searchProps={{
+              compressed: true,
+              ...(selectableProps ? selectableProps.searchProps : undefined),
```

Comment:
Are you wanting to allow `selectableProps.searchProps` to override the `compressed` prop??

Reply:
I suppose it would if the consumer also passed down a value in `selectableProps.searchProps.compressed`

Reply:
Ya as long as that is the intention 👍